qcodes/station.py

```diff
@@ -275,10 +275,9 @@ def close_all_registered_instruments(self) -> None:
         The instruments will stay registered as a component to the
         `Station`.
         """
-        for k, v in self.components.items():
-            if isinstance(v, Instrument):
-                inst = v
```

Comment:
Out of curiosity, why do `inst = v` and then `inst.close()` rather than just `v.close()`?

Reply:
this is a style thing. "v" is just a value from a dictionary of components, hence it can be an instrument but also something else. after the isinstance part, the "v" is clarified to be an "inst" instrument, and this is expressed via defining a new variable because closing an instrument is a valid operation, while closing "a component" may not be valid because not all station components are expected to support the "close" method.